tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KonanCompileTask.kt

```diff
@@ -201,6 +201,15 @@ open class KonanCompileConfig(
 
     // DSL. Libraries.
 
+    fun library(project: Project) = run {
+        compilationTask.dependsOn(project.tasks.findByPath("compileKonan"))
+        libraries(project.fileTree(project.konanCompilerOutputDir).include("*.klib"))
+    }
+    fun library(project: Project, artifactName: String) = run {
+        compilationTask.dependsOn(project.tasks.findByPath("compileKonan${artifactName.capitalize()}"))
```

Comment:
I think it's better to use a property of `KonanCompileConfig` instead of the task name:
```
compilationTask.dependsOn(project.konanArtifactsContainer.getByName(artifactName).compilationTask)
```
Also such access to a task of `project` will fail if `project` is not evaluated. So it's worth using the `evaluationDependsOn` method to be sure that `project` is already evaluated. It may look like this:
```
    fun library(project: Project, artifactName: String) {
        this.project.evaluationDependsOn(project.path)
        compilationTask.dependsOn(project.konanArtifactsContainer.getByName(artifactName).compilationTask)
        library("${project.konanCompilerOutputDir}/$artifactName.klib")
    }
```